pysal/contrib/viz/mapping.py

```diff
@@ -610,11 +611,17 @@ def geoplot(db, col=None, palette='BuGn', classi='Quantiles',
                   Additional named vaues to be passed to the classifier of choice.
     '''
     if col:
+        if hasattr(palette, 'number') and 'k' not in kwargs:
```

Comment:
This is the bit I'm not sure would work. If `k` was set and `palette` implicitly required a different `k`, I think it'd fail.